presto-spi/src/main/java/io/prestosql/spi/connector/TableToken.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.connector;
+
+public interface TableToken
```

Comment:
Does this need to be long term, stable, serializable?

Reply:
The idea is that the connector provides the token for each base table when the matview is refreshed. This token needs to be stored somewhere (mat view properties or in the snapshot of storage table if it is an iceberg table). So, you are right this token needs to be serializable.
Since the engine does not interpret the token and simply hands back to the connector to check if view is current, I think the versioning of the token can be managed by the connector. Is this what you meant by 'long term, stable'?